### PR TITLE
Add TENSILE_ENABLE_MARKER and HIPBLASLT_ENABLE_MARKER for C API

### PR DESCRIPTION
**Pull request by KKyang**
TENSILE_ENABLE_MARKER default is off.

HIPBLASLT_ENABLE_MARKER default is on.

TENSILE_ENABLE_MARKER enables markers inside Tensile lib.

HIPBLASLT_ENABLE_MARKER enables markers for hipBLASLt API calls.

### CODE REVIEW

**Comment by KKyang** [comment]
[----------] Global test environment tear-down
[==========] 48206 tests from 13 test suites ran. (3365716 ms total)
[  PASSED  ] 48206 tests.
hipBLASLt version: 1000

**Comment by KKyang** [comment]
===================================== 75 passed, 31 skipped, 48 warnings in 8677.11s (2:24:37) ======================================
py3: exit 0 (8677.37 seconds) /data0/yangwen/hipBLASLt/tensilelite> py.test -v --basetemp=/tmp/.tflite/py3/tmp --junit-xml=/data0/yangwen/hipBLASLt/tensilelite/python_tests.xml --junit-prefix=py3 --color=yes -n 4 --prebuilt-client=/tmp/.tflite/py3/client/0_Build/client/tensile_client Tensile/Tests -m common pid=1089288
  py3: OK (8820.89=setup[12.54]+cmd[0.71,130.26,8677.37] seconds)
  congratulations :) (8820.93 seconds)

File: library/CMakeLists.txt [hunk 1]
```diff
@@ -177,6 +177,8 @@ else()
   target_link_libraries(hipblaslt PRIVATE ${CUDA_cublaslt_LIBRARY})
 endif()
 
+target_link_libraries(hipblaslt PRIVATE  /opt/rocm/lib/libroctx64.so)
```
Comment (jichangjichang) [inline library/CMakeLists.txt]: follow ${ROCM_PATH} if it is set

**Review by jichangjichang — COMMENTED** [review]
Any other library enable roctx by default?

File: library/CMakeLists.txt (hunk 1 above)
Comment (KKyang) [reply library/CMakeLists.txt]: fixed

**Comment by KKyang** [comment]
> Any other library enable roctx by default?

CK?

**Comment by KKyang** [comment]
Comfirmed able to compile after the change

**Comment by jichangjichang** [comment]
> > Any other library enable roctx by default?
> 
> CK?

CK is not ROCM release library

**Comment by KKyang** [comment]
> > > Any other library enable roctx by default?
> > 
> > 
> > CK?
> 
> CK is not ROCM release library

SWDEV-433409

**Review by jichangjichang — COMMENTED** [review]
Need to add dependency with roctracer.
It also need to add the roctracer as dependent lib into ReadMe  and documentation.

**Review by jichangjichang — COMMENTED** [review]
it is also better to have build option for disabling/enabling the roctracer related implementation.

**Comment by KKyang** [comment]
> SWDEV-433409

Default on or off?

**Comment by KKyang** [comment]
> it is also better to have build option for disabling/enabling the roctracer related implementation.

Default on or off?

**Comment by KKyang** [comment]
Updated with options to turn on/off. Compile successfully.

**Comment by KKyang** [comment]
Change marker name "unload code object" to "UnloadCodeObjectFiles."

**Comment by KKyang** [comment]
Update doc and README.md

File: CMakeLists.txt [hunk 2]
```diff
@@ -239,6 +241,14 @@ else()
   find_package( hipblas-common REQUIRED CONFIG PATHS ${HIP_DIR} ${ROCM_PATH} /opt/rocm)
 endif()
 
+include(cmake/findRoctracer.cmake)
```
Comment (jichangjichang) [inline CMakeLists.txt]: Could you refer to https://github.com/ROCm/MIOpen/blob/61d9825a5eaec632a727e30d4ea881f28c38626f/CMakeLists.txt#L529-L530 and https://github.com/ROCm/MIOpen/blob/61d9825a5eaec632a727e30d4ea881f28c38626f/CMakeLists.txt#L533

File: library/CMakeLists.txt [hunk 3]
```diff
@@ -177,6 +177,8 @@ else()
   target_link_libraries(hipblaslt PRIVATE ${CUDA_cublaslt_LIBRARY})
 endif()
 
+target_link_libraries(hipblaslt PRIVATE -lroctx64)
```
Comment (jichangjichang) [inline library/CMakeLists.txt]: only link when tracer is enabled

File: tensilelite/Tensile/Source/lib/source/Debug.cpp [hunk 4]
```diff
@@ -27,6 +27,7 @@
 #include <Tensile/Debug.hpp>
 
 #include <mutex>
+#include <roctracer/roctx.h>
```
Comment (jichangjichang) [inline tensilelite/Tensile/Source/lib/source/Debug.cpp]: only include when tracer is enabled

File: CMakeLists.txt (hunk 2 above)
Comment (KKyang) [reply CMakeLists.txt]: Fixed

File: tensilelite/Tensile/Source/lib/source/Debug.cpp (hunk 4 above)
Comment (KKyang) [reply tensilelite/Tensile/Source/lib/source/Debug.cpp]: Fixed

File: CMakeLists.txt [hunk 5]
```diff
@@ -239,6 +241,14 @@ else()
   find_package( hipblas-common REQUIRED CONFIG PATHS ${HIP_DIR} ${ROCM_PATH} /opt/rocm)
 endif()
 
+if(HIPBLASLT_ENABLE_MARKER)
+  find_library(rocTracer roctx64)
+  if(NOT rocTracer)
+    message(FATAL_ERROR "roctracer not found, but HIPBLASLT_ENABLE_MARKER is enabled")
+  endif()
+  add_definitions(-DHIPBLASLT_ENABLE_MARKER)
```
Comment (jichangjichang) [inline CMakeLists.txt:249]: miss this part https://github.com/ROCm/MIOpen/blob/61d9825a5eaec632a727e30d4ea881f28c38626f/CMakeLists.txt#L533

Comment (KKyang) [reply CMakeLists.txt:249]: why we need this, we force the compiler stop instantly instead.

Comment (KKyang) [reply CMakeLists.txt:249]: I don't think we need this since we only use roctx

Comment (KKyang) [reply CMakeLists.txt:249]: Done by adding rocm_package_add_dependencies instead

File: library/src/amd_detail/rocblaslt/src/Debug.cpp [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+
+#include <Debug.hpp>
+
+#include <mutex>
+#include <roctracer/roctx.h>
```
Comment (jichangjichang) [inline library/src/amd_detail/rocblaslt/src/Debug.cpp]: include header when marker is enabled

Comment (KKyang) [reply library/src/amd_detail/rocblaslt/src/Debug.cpp]: Fixed

**Review by jichangjichang — DISMISSED** [review]
could we use inline for markerStart() and makerStop()

**Comment by KKyang** [comment]
> could we use inline for markerStart() and makerStop()

Yes, but clang does not guaranteed the function will be inlined.

**Review by jichangjichang — APPROVED** [review]
lgtm

**Review by hcman2 — APPROVED** [review]

**Pull request merged**